src/com/amazon/ionschema/internal/SchemaImpl.kt

```diff
@@ -129,35 +128,51 @@ internal class SchemaImpl private constructor(
                            header: IonStruct): Map<String, Import> {
 
         val importsMap = mutableMapOf<String, SchemaAndTypeImports>()
+        val importSet: MutableSet<String> = schemaSystem.getSchemaImportSet()
+
         (header.get("imports") as? IonList)
             ?.filterIsInstance<IonStruct>()
             ?.forEach {
                 val id = it["id"] as IonText
-                val importedSchema = schemaSystem.loadSchema(id.stringValue())
-                val schemaAndTypes = importsMap.getOrPut(id.stringValue()) {
-                    SchemaAndTypeImports(id.stringValue(), importedSchema)
-                }
+                val alias = it["as"] as? IonSymbol
+                // if importSet has an import with this id then do not load schema again to break the cycle.
+                if(!importSet.contains(id.stringValue())) {
+                    var normalizedId = schemaId?: "";
```

Comment:
Since we're no longer allowing/handling aliases, do we still need a `normalizedId`? Maybe instead of `normalizedId` and `id`  we could use names like `parentImportId` and `childImportId`?